src/driver/sqlserver/SqlServerQueryRunner.ts

```diff
@@ -282,28 +282,27 @@ export class SqlServerQueryRunner extends BaseQueryRunner implements QueryRunner
 
         request.query(query);
 
-        // Any event should release the lock.
+        const streamRequest = request.toReadableStream();
+
+        // Any event on the underlying request should release the lock.
         request.once("row", release);
         request.once("rowsaffected", release);
         request.once("done", release);
         request.once("error", release);
```

Comment:
I'm a bit unsure about these lines. If the lock is on the connection via a pool, then really we shouldn't be releasing it until the request has properly ended

Reply:
Probably we need to ask author of these changes (hope it's not me).

Reply:
After thinking about it, these should be removed and likely were there because there wasn't a proper stream being exposed in by of mssql, so this was just a bit of a "hack" to get it working

Reply:
I've made the change but also looked to track down the original author. This was added in https://github.com/typeorm/typeorm/pull/7893 by @imnotjames.

I'm a bit cautious about this particular part of the changes as I don't understand why a lock is even necessary (I've looked at the MySQL implementation and there is no equivalent lock happening on queries). So some insight as to why the lock is needed and what it's achieving should shed light on when the lock should be released.

Reply:
I don't know either. Let's merge this PR for now and keep track on @imnotjames response to your question.